spec/support/authentication.rb

```diff
@@ -50,3 +56,39 @@ def create_user(login)
   }
   let(:api_key) { the_user.credentials.api_key }
 end
+
+shared_context "create host" do
+  def create_host(host_login)
+    id = "rspec:host:#{host_login}"
+    host_role = Role.create(role_id: id).tap do |role|
+      options = { role: role }
+      Credentials.create(options)
+      role.reload
+    end
+    Resource.create(resource_id: id, owner_id: id).tap do |resource|
+      resource.reload
+      host_role.reload
+    end
+
+    return host_role
+  end
```

Comment:
This is a larger issue you couldn't possibly be expected to know about, especially as it's used in our own code bases, but I would consider `shared_context` in rspec to be an anti-pattern.  It is basically an rspec DSL reimplementation of a core ruby language feature, namely making a method or object available to some code via `require`.  This is perhaps a longer conversation but I'll just say that for now.

Reply:
i agree - this should probably be a larger conversation.